sdk/python/feast/permissions/security_manager.py

```diff
@@ -111,16 +110,17 @@ def assert_permissions_to_update(
     Raises:
         PermissionError: If the current user is not authorized to execute all the requested actions on the given resource or on the existing one.
     """
-    resource_to_update = assert_permissions(resource=resource, actions=actions)
+    actions = [AuthzedAction.DESCRIBE, AuthzedAction.UPDATE]
     try:
         existing_resource = getter(
-            name=resource_to_update.name,
+            name=resource.name,
             project=project,
             allow_cache=allow_cache,
         )  # type: ignore[call-arg]
         assert_permissions(resource=existing_resource, actions=actions)
     except FeastObjectNotFoundException:
-        pass
+        actions = [AuthzedAction.CREATE]
+    resource_to_update = assert_permissions(resource=resource, actions=actions)
```

Comment:
Shouldn't this always be a create action even if the object exists? `update` gives the authorization to overwrite something, while `create` describes what you can overwrite with... That way `update`, `delete`, `describe` actions will always be describing existing objects, while `create` will be applied to only the new one. idk, that makes more sense to me, wdyt?

Reply:
> while create describes what you can overwrite with
IMO create does not overwrite but generates a new instance. Maybe I'm not following your reasoning, could you give an example?

Reply:
I could, but never mind. this is more intuitive, thanks